feat: implement bhHiddenField

jniles opened the pull request:
This commit implements the `bhHiddenField` component which hides any transcluded HTML behind a toggle and lets the user decide when to show it and to hide it after.  The use case is for option fields in forms that do not need to be considered by default.  For example, many reports have complex options that should not be presented on the outset, but can instead by added in later.

Closes #933.

![clientsreportclosed](https://user-images.githubusercontent.com/896472/32426507-81521b88-c2c3-11e7-957f-bc9240fafa0d.PNG)
_Fig 1: Closed Hidden Field_

![clientsreportopen](https://user-images.githubusercontent.com/896472/32426508-82157466-c2c3-11e7-877f-009598a09204.PNG)
_Fig 2: Open Hidden Field_

jniles committed:
feat: implement bhHiddenField

This commit implements the `bhHiddenField` component which hides any
transcluded HTML behind a toggle and lets the user decide when to show
it and to hide it after.  The use case is for option fields in forms
that do not need to be considered by default.  For example, many reports
have complex options that should not be presented on the outset, but can
instead by added in later.

Closes #933.

DedrickEnc reviewed (commented):
@jniles

This feature is very important, it avoids information pollution in the system which very crucial for novices in computer system using (Vanga and Lukonga users). 
Information pollution can easily unoriente users.

However, it will be good to see a test for this component. It should expose a method to toggle the view for viewing/hiding components which are inside.

Great idea!

DedrickEnc commented on client/src/js/components/bhHiddenField.js:
```diff
@@ -0,0 +1,42 @@
+ */
+
+var template =
+ '<p>' +
```
This is so small but since we are not using ECMA 6 syntax to make string templating at the client side, you can create a separate file to make it easy to read.

jniles committed:
test(components): add bhHiddenField unit tests

This commit adds unit tests for the `bhHiddenField` component.  The
component is relatively simple - is only shows and hides transcluded
data.  These tests assert that the data is hidden by default and visible
when toggled on.

DedrickEnc reviewed (approved):
LGTM
